### DISCUSSION TITLE
mini.surround: how to assume motions if none are specified?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [nvim-mini/mini.nvim#1412] [General]
for instance, I want to write `sa"` in normal mode (no motions) and have it assume `iw` (like `saiw"`) or whatever I write in a config option for a default motion.

**Comment by echasnovski** [comment]
I am afraid, this is not how operators (and `sa` is a regular operator) in Neovim work. They always assume that next typed characters define textobject or motion to act upon. So in this case Neovim will treat `"` as textobject/motion and stop early because it (presumably) doesn't know textobject/motion with that start with that character.

The best you can do is to create custom Normal mode mappings for some common surroundings individually. So something like this:

```lua
-- **Do not** forget the `remap = true` part, it makes it treat keys "as if typed by user"
vim.keymap.set('n', 'sa"', 'saiw"', { remap = true })
```
